src/Common/src/System/Net/SocketAddress.cs

```diff
@@ -229,18 +229,50 @@ public override int GetHashCode()
 
         public override string ToString()
         {
-            var sb = new StringBuilder().Append(Family.ToString()).Append(':').Append(Size).Append(":{");
+            // Get the address family string.  In almost all cases, this should be a cached string
+            // from the enum and won't actually allocate.
+            string familyString = Family.ToString();
 
+            // Determine the maximum length needed to format.
+            int maxLength =
+                familyString.Length + // AddressFamily
+                1 + // :
+                10 + // Size (max length for a positive Int32)
+                2 + // :{
+                (Size - DataOffset) * 4 + // at most ','+3digits per byte
+                1; // }
+
+            Span<char> result = maxLength <= 256 ? // arbitrary limit that should be large enough for the vast majority of cases
+                stackalloc char[256] :
```

Comment:
Is this the recommended pattern for `stackalloc` or can we pass in `maxLength` instead of always allocating the worst case size of 256?

Reply:
> Is this the recommended pattern for stackalloc

Yes.  stackalloc'ing a constant is faster than stackalloc'ing a variable.